core/src/main/scala/matryoshka/Nu.scala

```diff
@@ -20,25 +20,28 @@ import Recursive.ops._
 
 import scalaz._, Scalaz._
 
-final case class Nu[F[_]](unNu: Id ~> F, a: Id[_])
+/** This is for coinductive (potentially infinite) recursive structures, models
+  * the concept of “codata”, aka, the “greatest fixed point”.
+  */
+sealed abstract class Nu[F[_]]
+private final case class Nu0[F[_], A](unNu: A => F[A], a: A) extends Nu[F]
```

Comment:
This is how you hide the `A`. I’m open to a better encoding of it, but this is how I figured out[†] how to do GADTs in Scala.

[†]: Read a few lines of a gist.